### ISSUE TITLE
docs: Add ABCI++ tutorials

### ISSUE THREAD

**Issue by thanethomson** [tendermint/tendermint#9272]
I've split out this issue from #9053. We need to add the following tutorials relating to ABCI++ before v0.37.0 lands:

- [x] `getting-started.md` [small, 0.5 days]
- [x] ~`indexing-transactions.md` [small, 0.5 days]~
- [x] `abci-cli.md` (review only) [small, 0.5 days]
- [x] `what-is-tendermint.md` (ABCI section) [small, 0.5 days]
- [x] `go-built-in.md and go.md` (same changes) [medium, 2 days]
- [x] ~new go-abci-plus-plus tutorial to demonstrate advanced ABCI++ usage~
  - ~for the time being, just PrepareProposal/ProcessProposal~

**Comment by cmwaters** [comment]
Some of these pages like `indexing-transactions` I don't think need touching. Sergio, Samantha and myself have had some prior discussion on working on the new ABCI docs so perhaps we can be assigned that work

**Comment by sergio-mena** [comment]
Right. File `indexing-transactions` will need to be changed for v0.38.x when we introduce `FinalizeBlock`, but not now. Let's leave it in this list though (I'll strike it through) so we can reuse this list for tracking our work on v0.38.x

**Comment by sergio-mena** [comment]
About assigning, I believe each of these could be picked up in parallel, so feel free to (a) assign this issue to yourself (you'd be picking all files), or (b) to hit the "convert to issue" button above for those files you want to pick up.

In the second case, please don't forget to:
* Assign the child issue to yourself
* Set the project to Tendermint, so that other members of the team (and of the community) have visibility on what's currently going on
* Set the status to "in progress"

**Comment by cmwaters** [comment]
I was thinking about this a little more. I know previously we had thought about writing a guide to building a public forum, but I think taking dydx's use case and writing a guide to an orderbook with an oracle might better demonstrate ABCI++ functionality

**Comment by sergio-mena** [comment]
Shouldn't we rather do that for v0.38.x, once we have vote extensions?
I think the work in this issue should rather be light... just adapting the text of those documents to the new API (only talking about `PrepareProposal`/`ProcessProposal`)

**Comment by cmwaters** [comment]
Well if you wanted to demonstrate a non-trivial case (i.e. where the txs are returned back to Tendermint), then an orderbook is a good example

**Comment by sergio-mena** [comment]
Good. In that case, please add one extra checkbox to this tracking issue: this extra checkbox should mention the creation of the tutorial on non-trivial usage of `PrepareProposal`/`ProcessProposal` (orderbook).

**Comment by lasarojc** [comment]
[PR 9688](https://github.com/tendermint/tendermint/pull/9688l) updates `go.md` and `go-built-in.md`.

**Comment by sergio-mena** [comment]
Closing this issue: as per conversation on Mon, the advanced ABCI++ tutorial is considered a stretch goal for this
